### PR TITLE
Move user settings config to an own section

### LINKED ISSUE #1812: Move docu + configuration for user settings
Currently the schema for usersettings and all custom functionality is handled in the settings section of luigi configuration.
We need to find a better place, it's not the best one.
Think about it and let's discuss in an after scrum session.

Important: Implementation should be backwards compatible.

### PR DESCRIPTION
**Pull request by JohannesDoberer**
Fixes #1812

### CODE REVIEW

**Comment by JohannesDoberer** [comment]
Docu will be fixed in #1818

**Comment by stanleychh** [comment]
There are several terms use `usersettings`. I would suggest change them to `userSettings`.  With the camel case.

File: core/src/utilities/helpers/usersetting-dialog-helpers.js [hunk 1]
```diff
@@ -6,7 +6,8 @@ class UserSettingsHelperClass {
 
     processUserSettingGroups() {
         const userSettingGroups = [];
-        const userSettingsSchema = LuigiConfig.getConfigValue('settings.userSettings.userSettingGroups');
+        let userSettingsGroupsFromConfig = LuigiConfig.getConfigValue('userSettings.userSettingGroups');
```
Comment (stanleychh) [inline core/src/utilities/helpers/usersetting-dialog-helpers.js:9]:
```suggestion
        const userSettingsGroupsFromConfig = LuigiConfig.getConfigValue('userSettings.userSettingGroups');
```

File: core/src/utilities/helpers/usersetting-dialog-helpers.js [hunk 2]
```diff
@@ -6,7 +6,8 @@ class UserSettingsHelperClass {
...
+        const userSettingsSchema = userSettingsGroupsFromConfig ? userSettingsGroupsFromConfig : LuigiConfig.getConfigValue('settings.userSettings.userSettingGroups');
```
Comment (stanleychh) [inline core/src/utilities/helpers/usersetting-dialog-helpers.js:10]: Could you also please declare a variable for `LuigiConfig.getConfigValue('settings.userSettings.userSettingGroups')`?   Same as `userSettingsGroupsFromConfig`. 
That will be helpful to understand at first glance.

**Review by UlianaMunich — APPROVED** [review]
LGTM

File: core/src/utilities/helpers/usersetting-dialog-helpers.js (hunk 1 above)
Comment (JohannesDoberer) [reply core/src/utilities/helpers/usersetting-dialog-helpers.js]: 👍

File: core/src/utilities/helpers/usersetting-dialog-helpers.js (hunk 2 above)
Comment (JohannesDoberer) [reply core/src/utilities/helpers/usersetting-dialog-helpers.js]: 👍

**Review by stanleychh — APPROVED** [review]

**Pull request merged**